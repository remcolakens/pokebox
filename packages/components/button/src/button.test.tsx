import { render, screen } from '@testing-library/react';
import { Button } from './index';

test('Button renders with the provided text', () => {
	render(<Button>React</Button>);

	expect(screen.getByText('React')).toBeInTheDocument();
});

test('Button onClick event works as expected', () => {
	let clickCount = 0;
	render(<Button onClick={() => clickCount++}>Click me</Button>);

	const buttonElement = screen.getByText('Click me');

	buttonElement.click();
	expect(clickCount).toBe(1);
});

test('Button matches snapshot', () => {
	const { asFragment } = render(<Button>Snapshot Test</Button>);
	expect(asFragment()).toMatchSnapshot();
});
