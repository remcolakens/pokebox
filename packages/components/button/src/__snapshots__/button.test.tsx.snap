// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="font-bold focus:ring-4 text-xl rounded-lg mr-2 mb-2 focus:outline-none text-white bg-blue-700 hover:bg-blue-800 focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 px-5 py-2.5"
  >
    Snapshot Test
  </button>
</DocumentFragment>
`;
